cmd/rook/ceph/cleanup.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2020 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ceph
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	"github.com/rook/rook/cmd/rook/rook"
+	"github.com/rook/rook/pkg/util/flags"
+	"github.com/spf13/cobra"
+)
+
+var (
+	dataDirHostPath string
+)
+
+var cleanUpCmd = &cobra.Command{
+	Use:   "clean",
+	Short: "Starts the cleanup process on the disks after ceph cluster is deleted",
+}
+
+func init() {
+	cleanUpCmd.Flags().StringVar(&dataDirHostPath, "data-dir-host-path", "", "dataDirHostPath on the node")
+	flags.SetFlagsFromEnv(cleanUpCmd.Flags(), rook.RookEnvVarPrefix)
+	cleanUpCmd.RunE = startCleanUp
+}
+
+func startCleanUp(cmd *cobra.Command, args []string) error {
+	rook.SetLogLevel()
+	rook.LogStartupInfo(cleanUpCmd.Flags())
+
+	logger.Infof("starting cluster clean up")
+
+	//Delete dataDirHostPath
+	if dataDirHostPath != "" {
+		if err := os.RemoveAll(fmt.Sprintf("%s/", dataDirHostPath)); err != nil {
+			return errors.Wrap(err, "failed to clean up dataDirHostPath")
+		}
+		logger.Info("successfully cleaned up dataDirHostPath")
+
+		// Remove all the mon directories.
+		parentDir := filepath.Dir(dataDirHostPath)
+		monDirs, err := filepath.Glob(fmt.Sprintf("%s/%s", parentDir, "mon-*"))
```

Comment:
How does using a glob ensure that we are **only** deleting **our** monitors and not another cluster monitors?

Reply:
If two clusters are using the same dataDirHostPath, the mons will already be colliding. It's just not supported to use the same host path.

Reply:
Not necessarily depending on the placement. `mon-a` from cluster1 and `mon-c` from cluster2 on the same host, no?

Reply:
Per latest discussion, if we have multiple clusters then they should be on different dataDireHostPaths.  So there can be no collision while deleting mon.  @leseb Please let me if there is still any disagreement on this.